src/web/app/components/dialogs/MaterialTestGeneratorPanel/TableSetting.ts

```diff
@@ -15,11 +15,22 @@ export interface Detail {
   selected: 0 | 1 | 2;
 }
 
+interface SettingInfos {
+  laserType?: LaserType;
+}
+
+type TableSettingConstruct<
+  T extends ReadonlyArray<string>,
+  IsRequired = true
+> = IsRequired extends true ? { [K in T[number]]: Detail } : { [K in T[number]]?: Detail };
+
 export const tableParams = ['strength', 'speed', 'repeat'] as const;
+export const promarkTableParams = ['fillInterval'] as const;
 export const mopaTableParams = ['pulseWidth', 'frequency'] as const;
```

Comment:
Allow Desktop Promark to test frequency, even though it may not have much influence
Also update `promarkTableSetting` in this file and `generateBlocks` in index.tsx